src/renderer/managers/plugins.ts

```diff
@@ -1,83 +1,44 @@
-  start: () => Promise<void>;
-  stop: () => Promise<void>;
-  runPlaintextPatches: () => Promise<void>;
-};
+interface PluginWrapper extends RepluggedPlugin {
+  exports: PluginExports | undefined;
+}
 
 /**
  * @hidden
  */
 export const plugins = new Map<string, PluginWrapper>();
-/**
- * @hidden
- */
-export const pluginExports = new Map<string, unknown>();
 
 const styleElements = new Map<string, HTMLLinkElement>();
 
 /**
- * Load a plugin
- * @param plugin Plugin class. You can get this from {@link get} or {@link list}
+ * Get the exports of a plugin.
+ * @param id Plugin ID
+ * @returns Exports of the plugin.
  *
  * @remarks
- * You may need to reload Discord after adding a new plugin before it's available.
+ * This is primarily intended to shorten plaintext patches that need to access exported
+ * functions or variables from their respective plugins.
+ * Instead of writing `replugged.plugins.plugins.get("id.here").exports`,
+ * developers can write `replugged.plugins.getExports("id.here")`.
  */
-export function load(plugin: RepluggedPlugin): void {
-  try {
-    let renderer: PluginExports;
-    const localExports: Record<string, unknown> = {};
-    pluginExports.set(plugin.manifest.id, localExports);
-    const pluginWrapper: PluginWrapper = Object.freeze({
-      ...plugin,
-      start: async (): Promise<void> => {
-        if (plugin.manifest.renderer) {
-          renderer = await import(
-            `replugged://plugin/${plugin.path}/${plugin.manifest.renderer}?t=${Date.now()}}`
-          );
-          await renderer.start?.();
-        }
-
-        const el = loadStyleSheet(
-          `replugged://plugin/${plugin.path}/${plugin.manifest.renderer?.replace(/\.js$/, ".css")}`,
-        );
-        styleElements.set(plugin.path, el);
-
-        log("Plugin", plugin.manifest.name, void 0, "Plugin started");
-      },
-      stop: async (): Promise<void> => {
-        await renderer?.stop?.();
-
-        if (styleElements.has(plugin.path)) {
-          styleElements.get(plugin.path)?.remove();
-          styleElements.delete(plugin.path);
-        }
-
-        log("Plugin", plugin.manifest.name, void 0, "Plugin stopped");
-      },
-      runPlaintextPatches: async () => {
-        if (typeof plugin.manifest.plaintextPatches === "string") {
-          patchPlaintext(
-            (
-              await import(
-                `replugged://plugin/${plugin.path}/${
-                  plugin.manifest.plaintextPatches
-                }?t=${Date.now()}`
-              )
-            ).default,
-          );
-        }
-      },
-    });
-    plugins.set(plugin.manifest.id, pluginWrapper);
-  } catch (e: unknown) {
-    error("Plugin", plugin.manifest.id, void 0, "Plugin failed to load\n", e);
+export function getExports(id: string): PluginExports | undefined {
+  const plugin = plugins.get(id);
+  if (plugin) {
+    return plugin.exports;
   }
+  throw new Error(`Plugin "${id}" does not exist or is not loaded`);
```

Comment:
IMO it's better to handle the edge cases first and then return the value at the end